nemo/collections/tts/g2p/models/ctc.py

```diff
@@ -437,3 +439,58 @@ def per(self):
     @per.setter
     def per(self, per):
         self._per = per
+
+    # Methods for model exportability
+    def _prepare_for_export(self, **kwargs):
+        super()._prepare_for_export(**kwargs)
+
+        # Define input_types and output_types as required by export()
+        self._input_types = {
+            "input_ids": NeuralType(('B', 'T'), TokenIndex()),
+            "input_len": NeuralType(tuple('B'), LengthsType()),
+        }
+        self._output_types = {
+            # "preds_str": NeuralType(('B', 'T'), LabelsType()),
+            "log_probs": NeuralType(('B', 'T'), LossType()),
```

Comment:
If string cannot be returned, then would it make sense to return the text/phoneme as indices? Which I guess would be the `greedy_predictions` output of `forward()`.

Reply:
investigated and as confirmed by @borisfom, decoding/encoding should happen outside of onnx model, so this looks correct. Also i am already using the onnx models build with this in productiuon, which i think confirms this is working ok.